src/Compilers/Core/Portable/Binding/BindingDiagnosticBag.cs

```diff
@@ -309,40 +309,40 @@ internal bool ReportUseSiteDiagnostic<TData>(DiagnosticInfo? info, Func<TData, L
         }
     }
 
-    internal readonly struct ImmutableBindingDiagnostic<TAssemblySymbol> where TAssemblySymbol : class, IAssemblySymbolInternal
+    internal readonly struct ReadOnlyBindingDiagnostic<TAssemblySymbol> where TAssemblySymbol : class, IAssemblySymbolInternal
     {
         private readonly ImmutableArray<Diagnostic> _diagnostics;
         private readonly ImmutableArray<TAssemblySymbol> _dependencies;
 
         public ImmutableArray<Diagnostic> Diagnostics => _diagnostics.NullToEmpty();
```

Comment:
We might want to go over usages of this API and make sure returned items aren't resolved without a good reason. For example:
- Binder_Statements.cs - ```hasErrors = constantValueDiagnostics.Diagnostics.HasAnyErrors();```